src/main/java/io/confluent/connect/jdbc/source/JdbcSourceTask.java

```diff
@@ -99,6 +99,8 @@ public void start(Map<String, String> properties) {
 
     cachedConnectionProvider = connectionProvider(maxConnAttempts, retryBackoff);
 
+    disableAutoCommit(cachedConnectionProvider.getConnection());
```

Comment:
should we not renable autocommit after every statement - isn't there times when we do need autocommit?

Reply:
In a single-threaded scenario, we could swap autocommit on and off. I initially thought it would be easier to follow if it was always disabled, and then do explicit commits everywhere.
But seeing as the only other query made in the task (other than the main query) is in `validateNonNullable`, which is only called during the start() method. I could move the disableAutoCommit after the validateNonNullable, and remove the explicit commit there.

I think after the task startup, it's simpler to leave autoCommit disabled, and then call commit explicitly at the conclusion of each query.